### PR TITLE
add support for configOverride property

### LINKED ISSUE #387: Provide a way to apply advanced configuration settings for Cassandra
**Is your feature request related to a problem? Please describe.**
k8ssandra currently exposes a limited set of Cassandra's configuration options from cassandra.yaml and from jvm.options. This is by design, and we will introduce support for more settings over time. There may be situations where users need to apply settings that are not exposed. 

Let's say I have run into a Cassandra bug that is causing major instability in my cluster. The work around involves applying some change in cassandra.yaml, but the property in question is not exposed. What are my options? I can directly edit the CassandraDatacenter to apply the change. This will work, but the change will be lost when I run a `helm upgrade`. That just compounds the problem.

**Describe the solution you'd like**
k8ssandra should provide some "hooks" to allow users to define arbitrary properties for cassandra.yaml and jvm.options. Since the configuration files are provisioned by cass-config-builder, it will be up to the user to understand what is supported and allowed by cass-config-builder. 

**Additional context**
A big motivation for this is customer support scenarios. We don't want this to turn into a backdoor that users rely on for all of their configuration. If that happens, then k8ssandra is falling short.

### PR DESCRIPTION
**Pull request by jsanda**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #387 

**Checklist**
- [x] Changes manually tested
- [x] Chart versions updated (if necessary)
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: charts/k8ssandra/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: k8ssandra
 description: |
   Provisions and configures an instance of the entire K8ssandra stack. This includes Apache Cassandra, Stargate, Reaper, Medusa, Prometheus, and Grafana.
 type: application
-version: 0.55.0
+version: 0.56.0
```
Comment (jakerobb) [inline charts/k8ssandra/Chart.yaml:6]: You'll need to bump to 57.

**Review by jakerobb — APPROVED** [review]
Looks good to me.

File: charts/k8ssandra/Chart.yaml (hunk 1 above)
Comment (jdonenine) [reply charts/k8ssandra/Chart.yaml:6]: Yeah we're definitely at 56 already, strange that it doesn't seem to show that here, hmm.

https://github.com/k8ssandra/k8ssandra/blob/df1f5f44925b247a483014176995d4d691ada417/charts/k8ssandra/Chart.yaml#L6

**Review by jdonenine — CHANGES_REQUESTED** [review]
Just need to bump the version and we're good.

**Review by emerkle826 — APPROVED** [review]

**Review by burmanm — APPROVED** [review]

**Review by jdonenine — APPROVED** [review]

**Pull request merged**